src/data/fin/vec_notation.lean

```diff
@@ -196,9 +196,9 @@ begin
   refl,
 end
 
-@[simp] lemma vec_append_apply_zero {α : Type*} {o : ℕ} (ho : (o + 1) = (m + 1) + n)
-  (u : fin (m + 1) → α) (v : fin n → α) :
-  vec_append ho u v 0 = u 0 := rfl
+@[simp] lemma vec_append_apply_zero {α : Type*} {o : ℕ} [ne_zero o] [ne_zero m] (ho : o = m + n)
+  (u : fin m → α) (v : fin n → α) : vec_append ho u v 0 = u 0 :=
+dif_pos _
```

Comment:
I think we should keep both versions of the lemma, since the original is a valid `dsimp` lemma. We have this elsewhere already, where the dsimp lemma has the primed name (`succ_zero_eq_one'`)

Reply:
I believe the thing that prompted this change is that the original lemma is no longer a `dsimp` lemma, so we'd remove it in the port anyway.

Reply:
Ah, I added a comment explaining why. I think we should still have both in lean3, and #noalign the dsimp lemma

Reply:
Really.
```lean
@[simp]
theorem vecAppend_apply_zero' {α : Type _} {o : ℕ} (ho : (o + 1) = (m + 1) + n)
    (u : Fin (m + 1) → α) (v : Fin n → α) : vecAppend ho u v 0 = u 0 :=
  rfl
```

```
type mismatch
  rfl
has type
  vecAppend ho u v 0 = vecAppend ho u v 0 : Prop
but is expected to have type
  vecAppend ho u v 0 = u 0 : Prop
```